### ISSUE TITLE
Support for TagSpaces Tagging Conventions in Files

### ISSUE THREAD

**Issue by Foadsf** [files-community/Files#15272]
### What feature or improvement do you think would benefit Files?

Integrating support for [TagSpaces tagging conventions](https://docs.tagspaces.org/tagging/) would significantly enhance the usability of Files for users who manage their files with TagSpaces tags. TagSpaces offers two tagging methods: an invasive method that incorporates tags directly into filenames within square brackets `[tag]`, and a non-invasive method that utilizes JSON files stored in a `.ts` directory adjacent to the tagged files or folders. Enabling Files to recognize and display these tags in its "info panel" would provide a seamless experience for users who rely on TagSpaces for file organization, making it easier to view and manage tagged files directly within Files. This is a follow-up to [this question](https://discordapp.com/channels/725513575971684472/725514946112127057/1234470765236322334) on Discord.

### Requirements

- Implement a parser in Files that can recognize and extract tags from filenames formatted according to TagSpaces' invasive tagging convention (tags within square brackets).
- Develop functionality to read and interpret JSON metadata files stored in .ts directories according to the non-invasive TagSpaces convention.
- Display the extracted tags in the Files' "info panel" for easy viewing.
- Ensure that tag data is updated in the Files interface when changes are made to filenames or JSON metadata files.
- Provide settings options in Files to enable or disable TagSpaces tag recognition, accommodating users' preferences.
- Conduct performance testing to ensure the new tagging features do not adversely affect the app’s performance.
- Create documentation and help guides to assist users in utilizing the new TagSpaces tagging features in Files.

### Files Version

3.3.0.0

### Windows Version

10.0.19045.4291

### Comments

This feature would be particularly beneficial for users who use TagSpaces extensively for file management but wish to utilize the sleek and powerful interface of Files for their daily tasks. It bridges the gap between two powerful tools, enhancing the organizational capabilities of Files while adhering to users' established workflows.

**Comment by Josh65-2201** [comment]
Thanks for the feedback, I've added this to the project board.

**Comment by yaira2** [comment]
How does TagSpaces work if a file is transferred or renamed?

**Comment by Foadsf** [comment]
@yaira2 Not sure, TBH. ¯\\\_(ツ)\_/¯ Asked them a question [here](https://x.com/fsfarimani/status/1785581286951670036) on Twitter, let's see if they respond.

**Comment by yaira2** [comment]
> If you rename/transfer such file outside of TagSpaces, the connection with meta file is lost, this happens also by the database based tagging solutions.

This is why we went with alternate data streams, they allow files to be moved externally without losing the tags.

**Comment by yaira2** [comment]
Correct, there are limitations with either approach, but we had to choose one.

From our docs (https://files.community/docs/features/tags)

> Tags are saved using alternate data streams (and not in a database). We found that alternate data streams is the best approach for Files because it allows items to be transferred between locations (even without using Files) without losing their tags. However, this system has some limitations:
> 
> Tags only work on drives formatted as NTFS
> Tags may not work with all cloud storage providers

**Comment by yaira2** [comment]
This feature request suggests that we provide an option for both methods. We'd prefer picking a single method but we'll leave this open for some time to gather additional feedback.

**Comment by yair100** [comment]
I have a proposal that should work

1. **Central Database Storage**: Implement a centralized database to store the paths of all tagged items. This will serve as the primary reference point for tagged data.
2. **Tag Embedding**: Embed the unique tag identifier (GUID) directly into the tagged file’s metadata. Note: This method is applicable exclusively to NTFS file systems due to their support for alternate data streams.
3. **Database Retrieval**: Upon initiating the tagged items’ loading process, the system will first retrieve the list from the central database. It will then apply a filter to exclude any items that are no longer accessible or have been removed.
4. **Search For Embedded Tags**: Files can then scan the file system for items containing tag-related alternate data streams. It will eliminate any duplicates already accounted for in the database to prevent redundancy.
